community_tasks/swiss_legal_evals.py

```diff
@@ -0,0 +1,402 @@
+# MIT License
+
+# Copyright (c) 2024 The HuggingFace Team
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# ruff: noqa: F405, F403, F401
+"""
+This module contains task configurations and prompt functions for evaluating
+LLM models on Swiss legal datasets. Each task is defined using the 
+`LightevalTaskConfig` class with its respective prompt function. The tasks 
+cover a variety of benchmarks, including: translation of laws, court decisions 
+and press releases.
+
+Author: Joel Niklaus
+"""
+from lighteval.logging.hierarchical_logger import hlog_warn
+import nltk
+
+nltk.download("punkt_tab")
+
+
+import numpy as np
+import re
+from dataclasses import dataclass
+
+
+from transformers import AutoTokenizer, AutoModelForSequenceClassification
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+from lighteval.metrics.metrics import Metrics
+from lighteval.metrics.metrics_sample import JudgeLLMMixEval, BertScore
+from lighteval.metrics.utils.metric_utils import (
+    MetricCategory,
+    MetricUseCase,
+    SampleLevelMetric,
+    SampleLevelMetricGrouping,
+)
+from lighteval.metrics.imports.bert_scorer import BERTScorer
+from lighteval.metrics.normalizations import remove_braces, remove_braces_and_strip
+from lighteval.tasks.extended.mix_eval.judge_prompts import (
+    flow_judge_for_freeform_template,
+    gpt_judge_for_closeended_freeform,
+)
+
+
+# CUSTOM METRICS
+def process_judge_response(x):
+    search = re.search(r"<score>\s(\d)\s</score>", x)
+    return int(search.group(1)) if search else 0
+
+
+def process_judge_response_freeform_gpt(x):
+    search = re.search(r"\[\[(\d.\d)\]\]", x)
+    answer = float(search.group(1) if search else 0)
+    return answer
```

Comment:
Could use a small explanation of the regex

Reply:
I copied this from https://github.com/huggingface/lighteval/blob/main/src/lighteval/tasks/extended/mix_eval/main.py.

Reply:
Why not import it from there directly then? :)

Reply:
Good point :)

Reply:
In general I was a bit unsure about how best to implement LLM judges. I want to use FlowAI, a closed LLM Judge and prometheus-eval. How would you recommend doing that?

Reply:
I'll let @NathanHB give you more detailed feedback on it when he comes back as he implemented this section :)

Reply:
Sounds good. Do you know when he is back? I am currently blocked by this.

Reply:
Well, the whole company's on an offsite till tomorrow, then coming back + jet lag, I'd say early next week.
In the meantime, did you take a look at the JudgeLM class [here](https://github.com/huggingface/lighteval/blob/main/src/lighteval/metrics/llm_as_judge.py) and the implementation of LLM as judge in [our MTBench](https://github.com/huggingface/lighteval/blob/main/src/lighteval/tasks/extended/mt_bench/main.py), which uses FlowAI (line 64)?

Reply:
Ok, thanks. Yes, I have seen those, but need to look into it in more detail. I was inspired by the mixeval implementation.

Reply:
Thanks so much! Is it intended to subclass JudgeLLM or instantiate it directly? I probably just didn't look at it in detail enough yet.

Now, I also just noticed that flow-judge and prometheus-eval have only been fine-tuned on English data. This means it does not make much sense to use them for these multilingual translation tasks. So I will just stick with GPT-4o evaluation.

Reply:
You can do both depending on what you need